Tools/Scripts/layout-test-failure-analysis

```diff
@@ -0,0 +1,291 @@
+    input_group.add_argument('-f', '--file',
+                        dest='input_file',
+                        help='Specifies the file to read for layout test results data.',
+                        type=str)
+    input_group.add_argument('-u', '-url',
+                        dest='input_url',
+                        help='Specifies the URL from which to retrieve layout test results data.',
+                        type=str)
+
+    # optional flags
+    parser.add_argument('-t', '--test-path',
+                        dest='test_path',
+                        help='Name of the test for which to search. Omitting the test name will export all workers as separate files (unless --combine-workers is used).',
+                        default='',
+                        type=str)
+    parser.add_argument('-o', '--output-path',
+                        dest='output_path',
+                        help='Specifies the file name for the tests output file. Defaults to run_tests.txt in current directory. If there will be multiple output files (workers), this will be treated as a directory.',
+                        default='run_tests.txt',
+                        type=str)
+    parser.add_argument('-l', '--long',
+                        action='store_true',
+                        help='Lengthens the worker output such that it doesn\'t truncate at the requested test. Good for getting a full test list. This has no effect when the test name is omitted.',
+                        required=False)
+    parser.add_argument('--stdio-export-path',
+                        dest="stdio_export_path",
+                        help='Specifies a path to which to save the raw stdio received from the server. This has no effect when loading data from a file.',
+                        required=False)
+    parser.add_argument('--combine-workers',
+                        action='store_true',
+                        help='Exports one file containing all tests in the order they were run (regardless of worker number).',
+                        required=False)
+    parser.add_argument('--no-warn',
+                        action='store_true',
+                        help='Does not warn or prompt the user before performing any actions. Potentially destructive due to bypassing file overwrite prompts. Good for automation. Will cause the script to quit if user input required.',
+                        required=False)
+
+    return parser.parse_args()
+
+
+# logging shortcuts
+def info(msg):
+    logging.info(msg)
+def warning(msg):
+    logging.warning(f'WARNING: {msg}')
+def error(msg):
+    logging.error(f'ERROR: {msg}')
+
+
+def graceful_exit(code=0):
+    print()
+    sys.exit(code)
+
+
+# Ensures a directory exists by checking and attempting to create if it doesn't exist (recursively).
+# Returns true if it exists at the end of the attempt, false if not.
+def ensure_directory_exists(dir):
+    if not os.path.isdir(dir):
```

Comment:
Why do we need this check? Seems like `os.makedirs` with `exist_ok` would imply this, no?

Would also simplify this function because you could imiediately just `return os.path.isdir(dir)`.

Reply:
This idea is that it attempts to make a directory and returns true/false depending on if it passes/fails. My thinking is this prevents fringe errors with permissions that (if I'm not mistaken) don't get caught by `os.path.isdir(dir)`.

Reply:
Good point!